### PR TITLE
Expand Initialization Support

### LINKED ISSUE #383: Expand Initializers Support
## Desired Behavior

Add support for startup scripts to load all data Netbox supports.

## Contrast to Current Behavior

I think only the circuits models are not supported.  IE CircuitType, Circuit, CircuitTermination, Provider.

## Changes Required

Create startup scripts and initializer yamls for the missing models.

## Discussion: Benefits and Drawbacks

Benefits include a more complete set of initializers that NetBox supports.  This allows developers, testers, and prod deployments to more fully leverage initializers for getting a deployment up and running.

Drawbacks really are the same as any startup script included in this project, IE that the project needs to update to match the  data model and that its all not necessarily within the core function of what the docker project is focused on.

### PR DESCRIPTION
**Pull request by ryanmerolle**
Related Issue: #383 

## New Behavior

Add additional Startup Scripts and example Initialization YAMLs

## Contrast to Current Behavior

New Feature expanding Startup Scripts Function

## Discussion: Benefits and Drawbacks

See linked Issue / Feature Request

## Changes to the Wiki

Current Wiki entry pretty much covers these new additions

## Proposed Release Note Entry

Added new startup scripts for [insert items once we finalize]

## Double Check

* [X] I have read the comments and followed the PR template.
* [X] I have explained my PR according to the information in the comments.
* [X] My PR targets the `develop` branch.

### CODE REVIEW

**Comment by ryanmerolle** [comment]
Need to think about how to handle option_assoc for 1 to many associations added for VRFs and Route Targets (for import and export targets):
```
 - enforce_unique: true
   name: vrf1
   tenant: tenant1
   description: main VRF
  import_targets:
   - "65000:1001"
   - "65000:1002"
  export_targets:
   - "65000:2001"
   - "65000:2002"
```

The same problem exists for tagged_vlans being associated to interfaces.

We likely need to recognize the key as a list and loop through each item for querying the association.

**Comment by ryanmerolle** [comment]
PR has been tested with all sample data.

Non-blocking nice to have items I leave up to @cimnine & @tobiasge to opine on are:
- Reorder startup_scripts to their liking for execution
- Add ability to setup 1 to many relationships for items like vrf import/export targets, and interface tagged vlans. (more detail above).
- Adding items from #383 that I mentioned, but did not add given ROI.

I will adjust based on feedback.

File: initializers/aggregates.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
-# - prefix: 10.0.0.0/16
-#   rir: RFC1918
-# - prefix: fd00:ccdd::/32
-#   rir: RFC4193 ULA
-# - prefix: 2001:db8::/32
-#   rir: RFC3849
+ # - prefix: 10.0.0.0/16
```
Comment (cimnine) [inline initializers/aggregates.yml]: What is the difference and why is it?

File: initializers/circuit_types.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+- name: VPLS
```
Comment (cimnine) [inline initializers/circuit_types.yml]: Please comment in all lines so that they aren't loaded by default.

File: initializers/power_feeds.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# - name: power feed 1
+#   #site: AMS 1
+#   power_panel: power panel AMS 1
+#   voltage: 208
+#   amperage: 50
+#   max_utilization: 80
+#   phase: Single phase
+#   rack: rack-01
+# - name: power feed 2
+#   #site: SING 1
```
Comment (cimnine) [inline initializers/power_feeds.yml]: if `site` is allowed, it'd be great to have one `site` which is not commented in order to test the scripts.

(Our test script removes `# ` from all the initializer yamls and then starts netbox to see if they all load without error.)

File: initializers/providers.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+- name: Provider1
```
Comment (cimnine) [inline initializers/providers.yml]: Please comment them in by default

File: initializers/services.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# - name: SSH
+#   protocol: TCP
+#   ports:
+#     - 22
+#   virtual_machine: virtual machine 1
+# - name: FTP
+#   protocol: TCP
```
Comment (cimnine) [inline initializers/services.yml]: Is UDP an option here? It's be great to know the options either by listing them as a comment or by showing them in various examples.

e.g. DNS is a UDP protocol (port 53, although it's also a TCP protocol.)

**Review by cimnine — COMMENTED** [review]
I still have to test it, but from what I've seen this looks solid. Thank you!

File: initializers/aggregates.yml (hunk 1 above)
Comment (ryanmerolle) [reply initializers/aggregates.yml]: Fixed whitespace mistakes.  I did not realize the tests remove the comments in the way you mentioned, so I was manually testing myself and made a mistake with adding comments back

File: initializers/providers.yml (hunk 4 above)
Comment (ryanmerolle) [reply initializers/providers.yml]: corrected

**Comment by cimnine** [comment]
There seems to be a problem with the power feed startup script or initializer yaml. To see it locally, run:

```bash
./build.sh master
./test.sh latest
```

See https://gist.github.com/cimnine/1201c3c93aa1ff1ded5d99f931a0b505

File: initializers/services.yml (hunk 5 above)
Comment (ryanmerolle) [reply initializers/services.yml]: Great idea.  I updated

**Comment by ryanmerolle** [comment]
I am assuming the 2 failed checks are unrelated to my PR

**Comment by cimnine** [comment]
> I am assuming the 2 failed checks are unrelated to my PR

Yes, they're unrelated.

**Review by cimnine — APPROVED** [review]
Thank you :)

**Comment by ryanmerolle** [comment]
> Thank you :)

Thanks for your assistance!

**Pull request merged**